src/pymovements/gaze/gaze_dataframe.py

```diff
@@ -1445,6 +1406,108 @@ def _fill_event_detection_kwargs(
+        if trial_columns is not None and len(trial_columns) == 0:
+            trial_columns = None
+        _check_trial_columns(trial_columns, self.frame)
+        self.trial_columns = trial_columns
+
+        # Initialize time column.
+        self._init_time_column(time_column, time_unit)
+
+        # Rename distance column if necessary.
+        if distance_column is not None and distance_column != 'distance':
+            self.frame = self.frame.rename({distance_column: 'distance'})
+
+        # Autodetect column names.
+        component_suffixes = ['x', 'y', 'xl', 'yl', 'xr', 'yr', 'xa', 'ya']
+
+        if auto_column_detect and pixel_columns is None:
+            column_canditates = ['pixel_' + suffix for suffix in component_suffixes]
+            pixel_columns = [c for c in column_canditates if c in self.frame.columns]
+
+        if auto_column_detect and position_columns is None:
+            column_canditates = ['position_' + suffix for suffix in component_suffixes]
+            position_columns = [c for c in column_canditates if c in self.frame.columns]
+
+        if auto_column_detect and velocity_columns is None:
+            column_canditates = ['velocity_' + suffix for suffix in component_suffixes]
+            velocity_columns = [c for c in column_canditates if c in self.frame.columns]
+
+        if auto_column_detect and acceleration_columns is None:
+            column_canditates = ['acceleration_' + suffix for suffix in component_suffixes]
+            acceleration_columns = [c for c in column_canditates if c in self.frame.columns]
+
+        # List of passed not-None column specifier lists.
+        # The list will be used for inferring n_components.
+        column_specifiers: list[list[str]] = []
+
+        # Nest multi-component columns.
+        if pixel_columns:
+            self._check_component_columns(pixel_columns=pixel_columns)
+            self.nest(pixel_columns, output_column='pixel')
+            column_specifiers.append(pixel_columns)
+
+        if position_columns:
+            self._check_component_columns(position_columns=position_columns)
+            self.nest(position_columns, output_column='position')
+            column_specifiers.append(position_columns)
+
+        if velocity_columns:
+            self._check_component_columns(velocity_columns=velocity_columns)
+            self.nest(velocity_columns, output_column='velocity')
+            column_specifiers.append(velocity_columns)
+
+        if acceleration_columns:
+            self._check_component_columns(acceleration_columns=acceleration_columns)
+            self.nest(acceleration_columns, output_column='acceleration')
+            column_specifiers.append(acceleration_columns)
+
+        self.n_components = self._infer_n_components(column_specifiers)
+
+    def _init_time_column(
+            self,
+            time_column: str | None = None,
+            time_unit: str | None = None,
+    ) -> None:
+        """Initialize time column."""
```

Comment:
args in docstring probably not needed due to private function? -- same question as above